src/site/twitch.tv/modules/settings/SettingsMenuButton.vue

```diff
@@ -11,37 +11,103 @@
 import Logo7TV from "@/assets/svg/logos/Logo7TV.vue";
 
+const updater = useUpdater();
+
 const emit = defineEmits<{
 	(event: "toggle"): void;
 }>();
 
-const updater = useUpdater();
+const props = defineProps<{
+	inst?: HookedInstance<Twitch.TopNavBarComponent>;
+}>();
+
+const hideBitsButton = useConfig<boolean>("layout.hide_bits_buttons");
+const hideSiteNotificationsButton = useConfig<boolean>("layout.hide_site_notifications_button");
+const hidePrimeCrown = useConfig<boolean>("layout.hide_prime_offers");
+const hideTurboButton = useConfig<boolean>("layout.hide_turbo_button");
+const hideWhispersButton = useConfig<boolean>("layout.hide_whispers_button");
+
+const menuElements = reactive({
+	renderBitsButton: hideBitsButton,
+	renderOnsiteNotificatons: hideSiteNotificationsButton,
+	renderTwitchPrimeCrown: hidePrimeCrown,
+	renderTurboButton: hideTurboButton,
+	renderWhispers: hideWhispersButton,
+});
+
+// force update top nav when user changes settings
+watch(menuElements, (renderElements) => {
+	rerenderTopNav(renderElements);
+});
+
+// TODO: define proper types here
+function rerenderTopNav(els) {
+	if (!props.inst?.component) return;
+	for (const renderFunc in els) {
+		defineFunctionHook(props.inst.component, renderFunc, function (old, ...args: unknown[]) {
+			return els[renderFunc] ? null : old?.apply(this, args);
+		});
+	}
+	toRaw(props.inst.component).forceUpdate();
+}
 
 const el = document.createElement("div");
 el.id = "seventv-settings-button";
 const containerEl = ref<HTMLElement>(el);
 
-// Topnav on the main twitch page
-let menuButtons = document.querySelector(".top-nav__menu")?.lastChild;
 let tooltip = "under";
+function insert7tvButton() {
+	const rootNode = props.inst?.domNodes.root;
+	let navmenu;
+	if (rootNode) {
+		// Place 7tv button next to the user profile
+		navmenu = rootNode.querySelector(".top-nav__menu")?.lastChild;
+	} else {
+		// Legacy
+		// Topnav on the main twitch page
+		navmenu = document.querySelector(".top-nav__menu")?.lastChild;
 
-if (!menuButtons) {
-	// Sidenav in modview
-	menuButtons = document.querySelector(".modview-dock__followed-channels")?.nextSibling;
-	tooltip = "right";
-}
-if (!menuButtons) {
-	// Header if popout
-	menuButtons = document.querySelector(".stream-chat-header")?.lastChild;
-	tooltip = "left";
-}
-if (menuButtons) {
-	menuButtons.insertBefore(containerEl.value, menuButtons.lastChild);
+		if (!navmenu) {
+			// Sidenav in modview
+			navmenu = document.querySelector(".modview-dock__followed-channels")?.nextSibling;
+			tooltip = "right";
+		}
+		if (!navmenu) {
+			// Header if popout
+			navmenu = document.querySelector(".stream-chat-header")?.lastChild;
+			tooltip = "left";
+		}
+	}
+	if (!navmenu) return;
+	if (navmenu.contains(containerEl.value)) {
+		// we have to re-insert the 7tv button since re-rendering will change order
+		navmenu.parentElement?.querySelector<HTMLElement>("#seventv-settings-button")?.remove();
```

Comment:
This is very strange and doesn't make sense, as well as being very dangerous if it worked as you expected it to (your modifying a vue element directly).

Why do this at all? You're traversing upwards only to query the element you already have a reference to, `containerEl.value` and `el` both refer to the element being queried here.